microsoft-edge/webview2/concepts/process-related-events.md

```diff
@@ -0,0 +1,293 @@
+
+##### [.NET/C#](#tab/dotnetcsharp)
+
+* [CoreWebView2BrowserProcessExitedEventArgs Class](https://learn.microsoft.com/dotnet/api/microsoft.web.webview2.core.corewebview2browserprocessexitedeventargs)
+
+##### [WinRT/C#](#tab/winrtcsharp)
+
+* [CoreWebView2BrowserProcessExitedEventArgs Class](https://learn.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2browserprocessexitedeventargs)
+
+##### [Win32/C++](#tab/win32cpp)
+
+* [ICoreWebView2BrowserProcessExitedEventArgs interface](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2browserprocessexitedeventargs)
+
+---
+
+
+<!-- ------------------------------ -->
+#### Clearing the user data folder
+
+Your application needs to wait until the WebView2 Runtime has released the _User Data Folder_ before it can delete its contents. After closing all WebView2 controls, the `BrowserProcessExited` event indicates this has happened and your application can proceed with the operation.
+
+See also:
+* [Manage user data folders](user-data-folder.md)
+
+
+<!-- ------------------------------ -->
+#### Updating the WebView2 Runtime
+
+In order to make use of the latest WebView2 Runtime after an update, your application needs to close all WebView2 controls and create a new `CoreWebView2Environment`. To ensure the new version is used, your application must wait for the `BrowserProcessExited` event; otherwise, the main browser process might stay alive when the new environment is created and switching to the new version would fail.
+
+
+<!-- ------------------------------ -->
+#### Restarting with a different environment configuration
+
+Most of the configuration used for a `CoreWebView2Environment` is bound to the main browser process lifetime. In order to make changes to this configuration (for example, to language), your application needs to close existing WebView2 controls and wait for `BrowserProcessExited` before recreating the controls; otherwise, initializing the WebView2 controls from the new CoreWebView2Environment might fail with incompatible configuration.
+
+
+<!-- ------------------------------ -->
+#### Clearing the auth cache
+
+The auth cache is bound to the main browser process lifetime. To clear the cache, your application must recreate its WebView2 controls from a new main browser process instance. To ensure a new main browser process instance is used when recreating the controls, your application must wait for the `BrowserProcessExited` event before proceeding; otherwise, the main browser process might stay alive when the controls are recreated and preserve the auth cache.
```

Comment:
@mikehoffms Should we incorporate the "definition" of auth cache here? Like:

"The auth cache, **which stores certificate selection and credentials from HTTPS Client Certificate Requests**, is bound to..."

Also, I'm not sure that the intention is to link to those pages related to auth cache. Those references were mostly for discussion about this article, not for public documentation.

Reply:
locally, changed to:
```
The _auth cache_ stores certificate selection and credentials from HTTPS Client Certificate Requests.  The auth cache is bound to the main browser process lifetime.  To clear the auth cache, your application must recreate its WebView2 controls from a new main browser process instance. To ensure that a new main browser process instance is used when recreating the controls, your application must wait for the `BrowserProcessExited` event before proceeding; otherwise, the main browser process might stay alive when the controls are recreated and preserve the auth cache.

<!-- 
* [Client Certificate Authentication](https://textslashplain.com/2020/05/04/client-certificate-authentication/)
* [With WebView2 what is the equivent of "closing the browser"?](https://github.com/MicrosoftEdge/WebView2Feedback/issues/2623) - WebView2Feedback repo issue #2623
-->
```